CRM/SMS/BAO/Provider.php

```diff
@@ -87,15 +88,20 @@ public static function getProviders($selectArr = NULL, $filter = NULL, $getActiv
   }
 
   /**
+   * Save a new record into the database
+   * @todo create a create function to do this work
    * @param $values
    */
   public static function saveRecord($values) {
+    $values['domain_id'] = CRM_Utils_Array::value('domain_id', $values, CRM_Core_Config::domainID());
```

Comment:
Would it be possible to set this in the civicrm_pre hook implrementation in the org.civicrm.multisite extension? This would go towards the goal of moving all multisite logic out of core.

Also Eileen suggested we might use an entity_domain table rather than adding domain columns in all these tables.